### PR TITLE
Add preset option for interests and skills (BACKEND)

### PR DESCRIPTION
**Pull request by dash102**
Backend of PR#83. Provides some starter constant presets for interests and skills

### CODE REVIEW

File: project/src/main/java/com/google/sps/utilities/PrefilledInformationConstants.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.google.sps.utilities;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class PrefilledInformationConstants {
```
Comment (liangqigong) [inline project/src/main/java/com/google/sps/utilities/PrefilledInformationConstants.java:7]: Let's add a comment on where do the labels come from.

File: project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet to get prefilled lists of information */
+@WebServlet("/prefilled-information")
+public class PrefilledInformationServlet extends HttpServlet {
```
Comment (liangqigong) [inline project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java:15]: Could you add a test for this?

File: project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+   *
+   * @param request servlet request - includes a 'category' parameter specifying the information to be returned
+   * @param response servlet response
+   * @throws IOException if Input/Output error occurs
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String category = request.getParameter("category");
+    List<String> information;
+    if (category.equals("interests")) {
```
Comment (liangqigong) [inline project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java:27]: Could you maybe throw an exception if it is neither interests nor skills? Usually we want to validate client's input if it is a enum value.

And use constants for interests and skills

File: project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java (hunk 2 above)
Comment (dash102) [reply project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java:15]: Added

File: project/src/main/java/com/google/sps/utilities/PrefilledInformationConstants.java (hunk 1 above)
Comment (dash102) [reply project/src/main/java/com/google/sps/utilities/PrefilledInformationConstants.java:7]: Done

File: project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java (hunk 3 above)
Comment (dash102) [reply project/src/main/java/com/google/sps/servlets/PrefilledInformationServlet.java]: Done

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+/** */
+@RunWith(JUnit4.class)
+public final class PrefilledInformationTest {
+  private static final String REQUEST_CATEGORY = "category";
+  private static final String INTERESTS = "interests";
+  private static final String SKILLS = "skills";
+
+  @Test
+  public void requestInterests() throws IOException {
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/PrefilledInformationTest.java:25]: You should lay out arrange, act and assert in the tests. And the same for the following one.

https://docs.google.com/document/d/1M9kKw-Qgwmi-fhIDab-HCsK6sp53qRJQaM7aM3s4rXs/edit#

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    PrintWriter writer = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(writer);
+    Mockito.when(request.getParameter(REQUEST_CATEGORY)).thenReturn(SKILLS);
+    new PrefilledInformationServlet().doGet(request, response);
+    writer.flush();
+    Assert.assertEquals(
+        CommonUtils.convertToJson(PrefilledInformationConstants.SKILLS).trim(),
+        stringWriter.toString().trim());
+  }
+}
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/PrefilledInformationTest.java:55]: Also test if it throws error if the request is neither SKILLS nor INTERESTS.

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+@RunWith(JUnit4.class)
+public final class PrefilledInformationTest {
+  private static final String REQUEST_CATEGORY = "category";
+  private static final String INTERESTS = "interests";
+  private static final String SKILLS = "skills";
+
+  @Test
+  public void requestInterests() throws IOException {
+    // Verify that returned interests is the provided interest constant
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/PrefilledInformationTest.java:27]: Move the common part to a SetUp funciton.
https://docs.google.com/document/d/1M9kKw-Qgwmi-fhIDab-HCsK6sp53qRJQaM7aM3s4rXs/edit#heading=h.9f87use2w5ox

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java (hunk 4 above)
Comment (dash102) [reply project/src/test/java/com/google/sps/PrefilledInformationTest.java:44]: Done - order is

mock getParameter return value
make GET call
assert

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java (hunk 5 above)
Comment (dash102) [reply project/src/test/java/com/google/sps/PrefilledInformationTest.java:70]: Done with Mockito verify

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java (hunk 6 above)
Comment (dash102) [reply project/src/test/java/com/google/sps/PrefilledInformationTest.java]: Done

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java (hunk 4 above)
Comment (liangqigong) [reply project/src/test/java/com/google/sps/PrefilledInformationTest.java:44]: You need to add empty lines among them, like:

```
mock getParameter return value

make GET call

assert
```

File: project/src/test/java/com/google/sps/PrefilledInformationTest.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  @After
+  public void flushWriter() {
+    Mockito.reset(request);
+    stringWriter.getBuffer().setLength(0);
+  }
+
+  @Test
+  public void requestInterests() throws IOException {
+    // Verify that returned interests is the provided interest constant
```
Comment (liangqigong) [inline project/src/test/java/com/google/sps/PrefilledInformationTest.java:45]: Move the comment  above the function

**Pull request merged**